python/equistore-operations/equistore/operations/equal.py

```diff
@@ -51,16 +51,20 @@ def equal_raise(tensor_1: TensorMap, tensor_2: TensorMap):
     :param tensor_1: first :py:class:`TensorMap`.
     :param tensor_2: second :py:class:`TensorMap`.
     """
-    try:
-        _check_same_keys(tensor_1, tensor_2, "equal")
-    except ValueError as e:
-        raise NotEqualError("the tensor maps have different keys") from e
+    message = _equal_impl(tensor_1=tensor_1, tensor_2=tensor_2)
+    if message != "":
+        raise NotEqualError(message)
+
+def _equal_impl(tensor_1: TensorMap, tensor_2: TensorMap) -> str:
+    message = _check_same_keys_impl(tensor_1, tensor_2, "equal")
+    if message != "":
+        return "The tensor maps have different keys: " + message
 
     for key, block_1 in tensor_1:
-        try:
-            equal_block_raise(block_1, tensor_2.block(key))
-        except NotEqualError as e:
-            raise NotEqualError(f"blocks for key '{key}' are different") from e
+        message = _equal_block_impl(block_1, tensor_2.block(key))
+        if message != "":
+            return f"blocks for key '{key}' are different: " + message
+    return ""
```

Comment:
```suggestion
        if message:
            return f"blocks for key '{key}' are different: " + message
    return ""
```

Reply:
Also, we could have a `NO_ERROR = ""` variable, and return that for better readability